File: remove_const.dart
import 'dart:io';

void main() {
  final dir = Directory('lib');
  final files = dir.listSync(recursive: true);
  for (var entity in files) {
    if (entity is File && entity.path.endsWith('.dart')) {
      var content = entity.readAsStringSync();
      var original = content;
      content = content.replaceAllMapped(RegExp(r'const\s+([A-Z])'), (m) => m.group(1)!);
      content = content.replaceAll(RegExp(r'const\s+\['), '[');
      content = content.replaceAll(RegExp(r'const\s+\{'), '{');
      if (content != original) {
        entity.writeAsStringSync(content);
      }
    }
  }
  print('Done');
}

File: remove_const.py
import os
import re

for root, dirs, files in os.walk('lib'):
    for file in files:
        if file.endswith('.dart'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # Replace 'const ClassName' -> 'ClassName'
            new_content = re.sub(r'const\s+([A-Z])', r'\1', content)
            
            # Replace 'const [' -> '['
            new_content = re.sub(r'const\s+\[', r'[', new_content)

            # Replace 'const {' -> '{'
            new_content = re.sub(r'const\s+\{', r'{', new_content)
            
            if new_content != content:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(new_content)
print("Done removing consts from constructors and collections.")
